hopr-socks/hopr-socks-server/tests/socks_server_tests.rs

```diff
@@ -85,26 +37,22 @@ async fn test_connect_client_incorrect_bind_address() {
             .expect("SOCKS5 server could not be run");
     });
 
-    sleep(Duration::from_millis(200)).await;
-
     // Open TCP connection through proxy (incorrect bind address)
-    let result = get_client(
-        String::from(bind_address),
-        String::from("127.0.0.2:1331"),
-        80,
-        AuthMode::NoAuth,
-    )
-    .await;
-    assert!(result.is_err());
+    let client = socks_client(String::from("127.0.0.2:1332"), AuthMode::NoAuth)
+        .await
+        .expect("Failed to create a SOCKS5 client");
+
+    assert!(get_request(&client, String::from("http://www.example.com"))
+        .await
+        .is_err());
 
     server_proc.abort();
-    sleep(Duration::from_millis(200)).await;
+    sleep(SERVER_DELAY).await;
```

Comment:
What is this server delay for?

Reply:
After server spawning, if you try right away to do a request proxy'ed by it, it most of the time fails. Seams that the server need a small time to be fully working.
Similarly, when shutting down the server, to avoid collisions for the next test, the delay ensures it's stopped in time.